### DISCUSSION TITLE
Java app using duckdb jdbc crashed with "terminate called without an active exception"

### DISCUSSION THREAD

**Discussion by fachen2015** [duckdb/duckdb#9345] [Q&A]
OS: centos 7.9
Java version: 1.8
duckdb jdbc version: 0.9.0

The Java app is multi-threaded, but each thread does only one job, it keeps opening local parquet file for reading using duckdb jdbc and processing the records afterwards. Each jdbc connection is cached in a thread local variable and is guaranteed to be used by the same thread, and its reinitialized only when SQLException happens (thread local is a tentative design to measure perf.)

Each connection is also initialized with S3 httpfs support, but so far only used on local files. The app could run happily for a while, then silently fail with the error. No java err logs, no cores generated.

Any help is much appreciated.

~Fang

**Comment by carlopi** [comment]
What if you remove the httpfs extension entirely? Does the error goes away?

**Reply by fachen2015** [reply]
Nope, removing the extension does not solve the issue.  I worked around it by using a jdbc thread pool instead of thread local based con cache and so far it runs fine.  I know this is a thread related issue, seems like duckdb internal thread was not happy about the way it got called and silently died, but I can't figure out why it happened though.
